### PR TITLE
[ new ] System.Random bindings

### PR DESCRIPTION
**Pull request by gallais**
Following a discussion on zulip

### CODE REVIEW

**Review by JacquesCarette — APPROVED** [review]
Nice.

Ah, good ole `trustMe`. But what's a little `trustMe` between `IO` friends, eh?

File: src/System/Random/Primitive.agda [hunk 1]
```diff
@@ -36,7 +36,7 @@ postulate
 {-# COMPILE GHC randomRIO-Int = randomRIO #-}
 {-# COMPILE GHC randomIO-Float = randomIO #-}
 {-# COMPILE GHC randomRIO-Float = randomRIO #-}
-{-# COMPILE GHC randomIO-Nat = randomIO #-}
+{-# COMPILE GHC randomIO-Nat = abs <$> randomIO #-}
```
Comment (gallais) [inline src/System/Random/Primitive.agda:39]: Nasty little bug here: `Nat` is `Integer` at runtime and so we need to be more careful here!

**Review by JacquesCarette — APPROVED** [review]

File: src/System/Random/Primitive.agda (hunk 1 above)
Comment (gallais) [reply src/System/Random/Primitive.agda:39]: If there ever were an ad for contributing a case to the testsuite when we mess with the FFI, that's it.

**Comment by MatthewDaggitt** [comment]
Looks great to me. No comments.

**Pull request merged**